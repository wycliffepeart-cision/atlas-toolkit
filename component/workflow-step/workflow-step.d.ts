export declare const WORKFLOW_STEP_ATTRIBUTES: string[];
export declare class WorkflowStep extends HTMLElement {
}
//# sourceMappingURL=workflow-step.d.ts.map