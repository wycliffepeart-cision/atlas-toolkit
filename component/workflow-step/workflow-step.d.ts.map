{"version":3,"file":"workflow-step.d.ts","sourceRoot":"","sources":["../../../src/component/workflow-step/workflow-step.ts"],"names":[],"mappings":"AAGA,eAAO,MAAM,wBAAwB,UAA6C,CAAC;AAEnF,qBACa,YAAa,SAAQ,WAAW;CAAG"}