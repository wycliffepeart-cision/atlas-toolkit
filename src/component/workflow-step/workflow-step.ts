import { Template } from '../../decorators/template';
import WorkflowStepTemplate from './workflow-step.html';

export const WORKFLOW_STEP_ATTRIBUTES = ['name', 'hero', 'active', 'hide-stepper'];

@Template(WorkflowStepTemplate)
export class WorkflowStep extends HTMLElement {}
