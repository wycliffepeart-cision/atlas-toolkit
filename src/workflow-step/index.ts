import { WorkflowStep } from './workflow-step';

if (!customElements.get('atlas-workflow-step')) customElements.define('atlas-workflow-step', WorkflowStep);
