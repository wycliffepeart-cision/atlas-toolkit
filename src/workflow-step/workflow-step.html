<template>
	<slot name="previousControls" part="previousControls"></slot>
	<slot name="nextControls" part="nextControls"></slot>

	<slot name="title" part="title"></slot>
	<slot name="subTitle" part="subTitle"></slot>
	<slot name="stepper" part="stepper"></slot>

	<slot part="body"></slot>

	<slot name="controls" part="controls"></slot>
</template>
