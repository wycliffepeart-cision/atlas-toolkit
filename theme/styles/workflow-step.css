@layer atlas {
	atlas-workflow-step {
		--step-gap: 2em;

		--step-title-font-size: 3em;
		--step-sub-title-font-size: 1em;
		--step-sub-title-max-width: 40ch;
		--step-title-font-weight: 700;

		--step-hero-title-font-color: var(--atlas-color-primary-b400);
		--step-hero-sub-title-font-size: 1.5em;
		--step-here-sub-title-font-weight: 400;

		--step-general-margin-inline: 1em;
	}

	atlas-workflow-step {
		min-height: 100%;
		width: 100%;

		display: flex;
		flex-direction: column;
		align-items: center;
		gap: var(--step-gap);

		padding-block-start: var(--step-gap);
	}

	atlas-workflow-step:where(:not([active]), [active="false"]) {
		display: none;
	}

	atlas-workflow-step::part(title) {
		display: block;

		font-size: var(--step-title-font-size);
		font-weight: var(--step-title-font-weight);

		margin-inline: var(--step-general-margin-inline);
	}

	atlas-workflow-step::part(subTitle) {
		font-size: var(--step-sub-title-font-size);
		margin-inline: var(--step-general-margin-inline);
	}

	atlas-workflow-step[hide-stepper]:not([hide-stepper="false"])::part(stepper) {
		display: none;
		justify-content: center;
	}

	atlas-workflow-step[hero]:not([hero="false"]) {
		background-image: var(--epic-hero-url, unset);
		background-size: cover;
	}

	atlas-workflow-step[hero]:not([hero="false"])::part(title) {
		color: var(--step-hero-title-font-color);
	}

	atlas-workflow-step[hero]:not([hero="false"])::part(subTitle) {
		font-size: var(--step-hero-sub-title-font-size);
		font-weight: var(--step-hero-sub-title-font-weight);
		text-align: center;

		max-width: var(--step-sub-title-max-width);
	}
}
