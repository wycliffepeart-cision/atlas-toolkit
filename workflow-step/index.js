import { WorkflowStep } from './workflow-step.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-workflow-step'))
    customElements.define('atlas-workflow-step', WorkflowStep);
