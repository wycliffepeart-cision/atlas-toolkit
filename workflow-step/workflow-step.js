import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var WorkflowStepTemplate = "<template>\n  <slot name=\"previousControls\" part=\"previousControls\"></slot>\n  <slot name=\"nextControls\" part=\"nextControls\"></slot>\n\n  <slot name=\"title\" part=\"title\"></slot>\n  <slot name=\"subTitle\" part=\"subTitle\"></slot>\n  <slot name=\"stepper\" part=\"stepper\"></slot>\n\n  <slot part=\"body\"></slot>\n\n  <slot name=\"controls\" part=\"controls\"></slot>\n</template>\n";

const WORKFLOW_STEP_ATTRIBUTES = ['name', 'hero', 'active', 'hide-stepper'];
let WorkflowStep = class WorkflowStep extends HTMLElement {
};
WorkflowStep = __decorate([
    Template(WorkflowStepTemplate)
], WorkflowStep);

export { WORKFLOW_STEP_ATTRIBUTES, WorkflowStep };
