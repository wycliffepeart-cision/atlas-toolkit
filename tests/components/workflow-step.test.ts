import '../../src/component/workflow-step/workflow-step';

const WORKFLOW_STEP_TAG = 'atlas-workflow-step';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Workflow Step', () => {
  it('is correctly appended in the DOM', () => {
    const elem = document.createElement(WORKFLOW_STEP_TAG);
    document.body.appendChild(elem);

    (expect(elem) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const elem = document.createElement(WORKFLOW_STEP_TAG);
    document.body.appendChild(elem);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });

  it('to respect having attributes ', () => {
    const elem = document.createElement(WORKFLOW_STEP_TAG);

    document.body.appendChild(elem);

    elem.toggleAttribute('hero', true);
    elem.toggleAttribute('active', true);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });
});
