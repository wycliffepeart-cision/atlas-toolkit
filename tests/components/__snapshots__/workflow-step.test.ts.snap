// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow Step is matches snapshot 1`] = `
<slot name="previousControls"
      part="previousControls"
>
</slot>
<slot name="nextControls"
      part="nextControls"
>
</slot>
<slot name="title"
      part="title"
>
</slot>
<slot name="subTitle"
      part="subTitle"
>
</slot>
<slot name="stepper"
      part="stepper"
>
</slot>
<slot part="body">
</slot>
<slot name="controls"
      part="controls"
>
</slot>
`;

exports[`Workflow Step to respect having attributes  1`] = `
<slot name="previousControls"
      part="previousControls"
>
</slot>
<slot name="nextControls"
      part="nextControls"
>
</slot>
<slot name="title"
      part="title"
>
</slot>
<slot name="subTitle"
      part="subTitle"
>
</slot>
<slot name="stepper"
      part="stepper"
>
</slot>
<slot part="body">
</slot>
<slot name="controls"
      part="controls"
>
</slot>
`;
